### ISSUE TITLE
Expand work on commit 4c92836, because it breaks bash tracing, as it has now

### ISSUE THREAD

**Issue by fgeorgatos** [plabrop/bic#64]
Case:
```
[fgeorgatos@node060 ~]$ bash -x -c 'echo hello'
+ set +x
hello
[fgeorgatos@node060 ~]$ unset BASH_ENV
[fgeorgatos@node060 ~]$ bash -x -c 'echo hello'
+ echo hello
hello
[fgeorgatos@node060 ~]$
```

Example how to save+restore xtrace (-x) option, which would be the better way to handle this:
```
xtrace_saved="$(set -o | grep xtrace | grep 'on')";
set +x

echo hello

[ -n "$xtrace_saved" ] && set -x 
unset xtrace_saved
```

**Comment by fgeorgatos** [comment]
ETA?!

**Comment by fgeorgatos** [comment]
@johnnydevaprasad : we need some progress on this one...

**Comment by fgeorgatos** [comment]
have a nice weekend all btw!

**Comment by fgeorgatos** [comment]
calling this fixed via #69

**Comment by fgeorgatos** [comment]
and #67
